### PR TITLE
Add resource group based LTR APIs + tests to azure sdk

### PR DESCRIPTION
**Pull request by pixia**

### CODE REVIEW

**Comment by jaredmoo** [comment]
Let's aim to publish this and https://github.com/Azure/azure-sdk-for-net/pull/6934 together

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 1]
```diff
@@ -221,6 +221,27 @@ public void TestLongTermRetentionV2Backups()
...
+            {
+                ResourceGroup resourceGroup = context.CreateResourceGroup();
+                Server server = context.CreateServer(resourceGroup);
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+                Database database = sqlClient.Databases.CreateOrUpdate(resourceGroup.Name, server.Name, SqlManagementTestUtilities.GenerateName(), new Database { Location = server.Location });
+
+                // Get the backups under the resource group, server and database. Assert there are no backups returned.
+                //
+                IPage<LongTermRetentionBackup> backups = sqlClient.LongTermRetentionBackups.ListByResourceGroupLocation(resourceGroup.Name, server.Location);
+                Assert.True(backups.Count() == 0);
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: removed the checks

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 2]
```diff
@@ -298,10 +319,10 @@ public void TestLongTermRetentionV2Crud()
             // Set the weekly retention on the database so that the first backup gets picked up
             // Wait about 18 hours until it gets properly copied and you see the backup when run get backups
             //
-            string locationName = "";
-            string resourceGroupName = "";
-            string serverName = "";
-            string databaseName = "";
+            string locationName = "brazilsouth";
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: discussed offline, added comments and keep the change so test could pass in playback mode

**Review by erich-wang — CHANGES_REQUESTED** [review]
Please resolve merge conflicts

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 3]
```diff
@@ -373,6 +394,83 @@ public void TestLongTermRetentionV2Crud()
...
 
+        [Fact(Skip = "Manual test due to long setup time required (over 18 hours).")]
+        public void TestLongTermRetentionV2ResourceGroupBasedCrud()
+        {
+            // MANUAL INSTRUCTIONS
+            // Create a server and database and fill in the appropriate information below
+            // Set the weekly retention on the database so that the first backup gets picked up
+            // Wait about 18 hours until it gets properly copied and you see the backup when run get backups
+            //
+            string locationName = "brazilsouth";
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: discussed offline will keep the change

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 4]
```diff
@@ -312,7 +333,7 @@ public void TestLongTermRetentionV2Crud()
                 //
                 Microsoft.Azure.Management.Sql.Models.BackupLongTermRetentionPolicy parameters = new Microsoft.Azure.Management.Sql.Models.BackupLongTermRetentionPolicy(weeklyRetention: "P2W");
                 var policyResult = sqlClient.BackupLongTermRetentionPolicies.CreateOrUpdateWithHttpMessagesAsync(resourceGroupName, serverName, databaseName, parameters).Result;
-                sqlClient.GetPutOrPatchOperationResultAsync(policyResult, new Dictionary<string, List<string>>(), CancellationToken.None).Wait();
+                Assert.Equal(System.Net.HttpStatusCode.OK, policyResult.Response.StatusCode);
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: Removed

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 5]
```diff
@@ -364,7 +385,7 @@ public void TestLongTermRetentionV2Crud()
                 //
                 parameters = new Microsoft.Azure.Management.Sql.Models.BackupLongTermRetentionPolicy(weeklyRetention: "P1W");
                 policyResult = sqlClient.BackupLongTermRetentionPolicies.CreateOrUpdateWithHttpMessagesAsync(resourceGroupName, serverName, databaseName, parameters).Result;
-                sqlClient.GetPutOrPatchOperationResultAsync(policyResult, new Dictionary<string, List<string>>(), CancellationToken.None).Wait();
+                Assert.Equal(System.Net.HttpStatusCode.OK, policyResult.Response.StatusCode);
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: Removed

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 6]
```diff
@@ -373,6 +394,83 @@ public void TestLongTermRetentionV2Crud()
...
+
+                // Delete the backup.
+                //
+                var deleteResult = sqlClient.LongTermRetentionBackups.DeleteByResourceGroupWithHttpMessagesAsync(resourceGroupName, locationName, serverName, databaseName, backup.Name).Result;
+                Assert.Equal(System.Net.HttpStatusCode.OK, deleteResult.Response.StatusCode);
+
+                // Verify the backup is gone.
+                //
+                backups = sqlClient.LongTermRetentionBackups.ListByResourceGroupDatabase(resourceGroupName, locationName, serverName, databaseName);
+                Assert.True(backups.Count() == 0);
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: Removed the check

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs [hunk 7]
```diff
@@ -373,6 +394,83 @@ public void TestLongTermRetentionV2Crud()
...
+                    {
+                        Location = locationName,
+                        CreateMode = CreateMode.RestoreLongTermRetentionBackup,
+                        LongTermRetentionBackupResourceId = backup.Id
+                    });
+
+                // Delete the backup.
+                //
+                var deleteResult = sqlClient.LongTermRetentionBackups.DeleteByResourceGroupWithHttpMessagesAsync(resourceGroupName, locationName, serverName, databaseName, backup.Name).Result;
+                Assert.Equal(System.Net.HttpStatusCode.OK, deleteResult.Response.StatusCode);
```
[the remark this answers is not in the document]
Comment (pixia) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/DatabaseRestoreScenarioTests.cs]: Yes, did that

**Comment by pixia** [comment]
Done

**Review by erich-wang — APPROVED** [review]

**Review by jaredmoo — APPROVED** [review]

**Pull request merged**